src/main/javascript/crypto/src/eas/AttestationUrl.ts

```diff
@@ -0,0 +1,312 @@
+
+import * as pako from "pako";
+import * as Base64 from "js-base64";
+import {SignedOffchainAttestation} from "@ethereum-attestation-service/eas-sdk";
+import {ethers} from "ethers";
+
+export interface SignedOffchainAttestationV1
+	extends Omit<SignedOffchainAttestation, "signature"> {
+	r: string;
+	s: string;
+	v: number;
+}
+
+export interface AttestationShareablePackageObject {
+	/** Signed typed data with attestation object */
+	sig: SignedOffchainAttestation | SignedOffchainAttestationV1;
+	/** Address of the signer */
+	signer: string;
+}
```

Comment:
@oleggrib I've left this out of the actual attestation since it is included in a separate magic link parameter like our ASN.1 tickets. I think it's best to keep it out as you don't necessarily want to provide the secret bundled with the attestation. 

Although I spoke to Tore and we could potentially provide the secret to the third party site to do a lightweight validation (user provides email address to check the commitment is valid) for specific use cases. i.e. Company A attestation provides a way for Company B to open accounts for users with a specific discount.

Reply:
@micwallace , for "secret" param we can skip it if EasTicketAttestation.getEasJson() returns different object than AttestationShareablePackageObject, thanks.

> we could potentially provide the secret to the third party site
it will work for commitment validation, but its not a very good idea to share original secret. better let third-party website create ID_attestation for this Email (just generate without signature and idAttestationSecret=hash(current time+requestor_domain)) and Outlet can respond with ZK-Proof of ticketSecret+idAttestationSecret

in that case: 
1. original secret stay secured
2. can check if that request was genereted in a last minute
3. can check domain(context), to avoid man-in-the-middle attack

Reply:
This is a great idea! Even better because it could potentially be done all within the outlet if no OTP is required. 

We should at least have a way of requesting ID_attestation without a wallet at least. Maybe by passing the address param as false.

@AW-STJ Let's consider this for future improvement.

Reply:
Thanks, @micwallace , we can update code and add an option for this flow. 
We dont need signature, we just need
- requestor domain
- request timestamp
- commitment

"requestor domain + request timestamp" will be transformed to secret

but if we want to stay flow same then one-time random wallet can be generated to sign ID Attestation